packages/neuron-wallet/src/services/tx/transaction-persistor.ts

```diff
@@ -342,25 +355,27 @@ export class TransactionPersistor {
   // when fetch a transaction, use TxSaveType.Fetch
   public static convertTransactionAndSave = async (
     transaction: Transaction,
-    saveType: TxSaveType
+    saveType: TxSaveType,
+    lockArgsSet?: Set<string>,
   ): Promise<TransactionEntity> => {
     const tx: Transaction = transaction
 
     let txEntity: TransactionEntity
     if (saveType === TxSaveType.Sent) {
       txEntity = await TransactionPersistor.saveWithSent(tx)
     } else if (saveType === TxSaveType.Fetch) {
-      txEntity = await TransactionPersistor.saveWithFetch(tx)
+      txEntity = await TransactionPersistor.saveWithFetch(tx, lockArgsSet)
     } else {
       throw new Error('Error TxSaveType!')
     }
     return txEntity
   }
 
-  public static saveFetchTx = async (transaction: Transaction): Promise<TransactionEntity> => {
+  public static saveFetchTx = async (transaction: Transaction, lockArgsSet?: Set<string>): Promise<TransactionEntity> => {
```

Comment:
For this function, it does not care whether the `lockArgsSet` is owned by itself.

Reply:
The call chain is as follows:
```
saveFetchTx -> TransactionPersistor.convertTransactionAndSave ->
saveWithFetch -> TransactionPersistor.create(..., lockArgsSet)
```
The parameter is used in `TransactionPersistor.create` as follows:
```
currentWalletInputs = inputs.filter(v => this.isCurrentWalletCell(v, lockArgsSet))
```
For the `isCurrentWalletCell(cell: InputEntity | OutputEntity, lockArgsSet: Set<string>)` function, the parameter should actually be `currentWalletLockArgsSet` (also known as `selfLockArgsSet`).

Therefore, the parameter required by `TransactionPersistor.create` should be `selfLockArgsSet` or an abstraction that implements `isSelfInputOrOutput`.

Following the call chain upwards, if the upper layers do not have branch checks such as `isSelf(lockArgsSet)`, then the meaning of `lockArgsSet` must be `selfLockArgsSet`.

Reply:
Maybe I need to rename `currentWalletInputs`? The `lockArgsSet` is from `queue`'s constructor's `addresses`, but the `addresses` is not named `selfAddresses`

Reply:
I understand now.

In this call chain (at least in `TransactionPersistor.create`), its meaning could be something like "lockArgsSet for which details need to be saved".

It is a broader abstraction than what `selfLockArgsSet` represents, so perhaps it could be named `saveDetailLockArgsSet`.

The names `isCurrentWalletCell` and `currentWalletInputs` do indeed need adjustment.

Reply:
Sounds good.

Reply:
How about `lockArgsSetNeedsDetail`, and `saveDetailLockArgsSet` like a verb that we will save detail?

Reply:
> How about `lockArgsSetNeedsDetail`

LGTM